xarray/plot/accessor.py

```diff
@@ -179,7 +182,7 @@ def step(self, *args, **kwargs) -> list[Line3D] | FacetGrid[DataArray]:
         return dataarray_plot.step(self._da, *args, **kwargs)
 
     @overload
-    def scatter(
+    def scatter(  # type: ignore[misc]  # None is hashable :(
```

Comment:
> I saw in the NamedArray PR we sometimes use this trick:
> I am not sure how scary it would be to just change to this method?

I don't think one should deviate from None as default if it is preventable.
This issue if exactly this: https://github.com/python/mypy/issues/13805
Not sure why for some methods it is only surfacing now and not already before.

Reply:
> Have you tried Optional[Hashable] ?

That is exactly the same as `Hashable | None`, and tools like pyupgrade will also replace that.

Reply:
I guess the question is; does xarray allow `None` as a dim/coordinate? If we do, we have to use other defaults.

Considering our insistence to use `Hashable` for dims it seems we want to:
```python
a = xr.Variable(data=[3,4], dims=(None,))
print(a)
<xarray.Variable (None: 2)>
array([3, 4])

c = xr.DataArray([1,2]).assign_coords(coords={"dim_0": [3,4]}).rename({"dim_0": None})
print(c)
<xarray.DataArray (None: 2)>
array([1, 2])
Coordinates:
  * None     (None) int32 3 4
```

I suppose this is just another issue with dim(s) haven't been consistently typed before.

Reply:
While None is technically hashable probably half of xarrays functions use it as default for dimension arguments, so it would be quite a large breaking change to allow it as a dimension/variable/coordinate "name".

I think it is not worth it to introduce such a breaking change just to allow such a corner case.
If something, we should probably add a check for not None in the constructors of xarray objects.

Reply:
If you want, you can open a follow up issue for that.
Definitely this is out of scope for this PR.